gandalf.js

```diff
@@ -1,92 +1,93 @@
-/**
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?